### PR TITLE
Add option to check share ACL's when listing directories

### PR DESCRIPTION
**Pull request by icewind1991**
Signed-off-by: Robin Appelman <robin@icewind.nl>

### CODE REVIEW

**Comment by Fonata** [comment]
1. I suggest to split this into topical commits. For example, I would prefer one for composer-related changes. 

2. Can you describe in more detail which install scenario and use case this PR solves? Thanks!

File: apps/files_external/lib/Lib/Storage/SMB.php [hunk 1]
```diff
@@ -133,7 +138,7 @@ public function __construct($params) {
 	private function splitUser($user) {
 		if (strpos($user, '/')) {
 			return explode('/', $user, 2);
-		} elseif (strpos($user, '\\')) {
+		} else if (strpos($user, '\\')) {
```
Comment (skjnldsv) [inline apps/files_external/lib/Lib/Storage/SMB.php:141]:
```suggestion
		} elseif (strpos($user, '\\')) {
```

**Review by skjnldsv — APPROVED** [review]
Conflicts :boom:

**Comment by icewind1991** [comment]
rebased, unit tests are happy

**Review by juliusknorr — APPROVED** [review]
Looks sane to me :+1:

**Comment by juliushaertl** [comment]
Failing php-cs check :wink:

**Pull request merged**